sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -515,6 +520,36 @@ class EthService(
     }
   }
 
+  @RpcMethod("debug_traceBlockByHash")
+  @RpcOptionalParameters(1)
+  def traceBlockByHash(hash: Hash, config: TraceOptions): DebugTraceBlockView = {
```

Comment:
Please refactor to not duplicate so much code. The only difference between `traceBlockByHash` and `traceBlockByNumber` is that the `byNumber` version needs to find the block hash via block-number or "tag" first, everything else is identical:
- trace-by-hash is `hash => traceBlock(hash)`
- trace-by-number is `tag => traceBlock(getBlockIdByTag(tag))`

Reply:
Agree, method `traceBlockByNumber` refactored